api/services/contact_fields_mapping_service.py

```diff
@@ -0,0 +1,107 @@
+
+    def handle_contact_groups_data(self, jsonData):
+        try:
+            self.set_init_data(jsonData)
+            user_details = models.User.query.get_by_phone(self.user_phone)
+            if not user_details:
+                return False
+
+            contact_groups = jsonData["contact"]["groups"]
+
+            for contact_group in contact_groups:
+                contact_group_name = contact_group["name"]
+                user_contact_group_mapping = (
+                    models.ContactFieldsMapping.query.get_by_group_name(
+                        contact_group_name
+                    )
+                )
+                if user_contact_group_mapping:
+                    self.process_contact_groups_data(
+                        user_contact_group_mapping, user_details
+                    )
+        except Exception as e:
+            print(f"Error occurred while handling custom field mapping. Exception: {e}")
+
+    def process_contact_fields_data(self, user_contact_field_details, user_details):
+        try:
+            for user_contact_field_detail in user_contact_field_details:
+                table_object = helpers.get_class_by_tablename(
+                    user_contact_field_detail.mapped_table_name
+                )
+                if table_object:
+                    column_name = user_contact_field_detail.mapped_table_column_name
+                    mapped_table_column_value = (
+                        user_contact_field_detail.mapped_table_column_value
+                    )
+                    self.update_mapped_fields(
+                        table_object,
+                        column_name,
+                        mapped_table_column_value,
+                        user_details,
+                    )
+        except Exception as e:
+            print(f"Exception occurred: {e}")
+
+    def process_contact_groups_data(self, user_contact_group_details, user_details):
+        try:
+            for user_contact_group_detail in user_contact_group_details:
+                table_object = helpers.get_class_by_tablename(
+                    user_contact_group_detail.mapped_table_name
+                )
+                if table_object:
+                    column_name = user_contact_group_detail.mapped_table_column_name
+                    mapped_table_column_value = (
+                        user_contact_group_detail.mapped_table_column_value
+                    )
+                    self.update_mapped_fields(
+                        table_object,
+                        column_name,
+                        mapped_table_column_value,
+                        user_details,
+                    )
+        except Exception as e:
+            print(f"Exception occurred: {e}")
+
+    def update_mapped_fields(
+        self, table_object, column_name, mapped_table_column_value, user_details
+    ):
+        try:
+            class_object_data = table_object.get_by_user_id(user_details.id)
```

Comment:
In case of user program table, what it will return?
```suggestion
            table_data = table_object.get_by_user_id(user_details.id)
```

Reply:
I think if a user has multiple user programs then I think this will return multiple data or incorrect data

Reply:
In the case of the user_program table get_by_user_id only returns the latest active user_program. So, it will not affect the other programs of user.